Allow deepseq-1.5

This would permit GHC 9.8.

Test suite passes with the bound bumped:

```
 % cd /tmp
janus@janus-ux305ca /tmp % cabal get psqueues
Unpacking to psqueues-0.2.7.3/
janus@janus-ux305ca /tmp % cd psqueues-0.2.7.3 
janus@janus-ux305ca /tmp/psqueues-0.2.7.3
 % cabal test --allow-newer=psqueues:deepseq
Warning: Unknown/unsupported 'ghc' version detected (Cabal 3.10.1.0 supports
'ghc' version < 9.8): /home/janus/.ghcup/bin/ghc is version 9.8.1
Resolving dependencies...
Build profile: -w ghc-9.8.1 -O1
In order, the following will be built (use -v for more details):
 - psqueues-0.2.7.3 (lib) (first run)
 - tasty-1.4.3 (lib) (requires build)
 - tasty-quickcheck-0.10.2 (lib) (requires build)
 - tasty-hunit-0.10.1 (lib) (requires build)
 - psqueues-0.2.7.3 (test:psqueues-tests) (first run)
Starting     tasty-1.4.3 (lib)
Configuring library for psqueues-0.2.7.3..
Building     tasty-1.4.3 (lib)
Preprocessing library for psqueues-0.2.7.3..
Building library for psqueues-0.2.7.3..
[1 of 7] Compiling Data.BitUtil     ( src/Data/BitUtil.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/BitUtil.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/BitUtil.dyn_o )
[2 of 7] Compiling Data.IntPSQ.Internal ( src/Data/IntPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/IntPSQ/Internal.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/IntPSQ/Internal.dyn_o )

src/Data/IntPSQ/Internal.hs:66:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
   |
66 | import           Control.Applicative ((<$>), (<*>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/IntPSQ/Internal.hs:70:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
   |
70 | import           Data.Foldable       (Foldable)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/IntPSQ/Internal.hs:75:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Word’ is redundant
      except perhaps to import instances from ‘Data.Word’
    To import instances alone, use: import Data.Word()
   |
75 | import           Data.Word           (Word)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[3 of 7] Compiling Data.IntPSQ      ( src/Data/IntPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/IntPSQ.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/IntPSQ.dyn_o )
[4 of 7] Compiling Data.OrdPSQ.Internal ( src/Data/OrdPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/OrdPSQ/Internal.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/OrdPSQ/Internal.dyn_o )
[5 of 7] Compiling Data.OrdPSQ      ( src/Data/OrdPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/OrdPSQ.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/OrdPSQ.dyn_o )
[6 of 7] Compiling Data.HashPSQ.Internal ( src/Data/HashPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/HashPSQ/Internal.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/HashPSQ/Internal.dyn_o )

src/Data/HashPSQ/Internal.hs:59:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
   |
59 | import           Data.Foldable        (Foldable)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[7 of 7] Compiling Data.HashPSQ     ( src/Data/HashPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/HashPSQ.o, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/build/Data/HashPSQ.dyn_o )
Installing   tasty-1.4.3 (lib)
Completed    tasty-1.4.3 (lib)
Starting     tasty-quickcheck-0.10.2 (lib)
Starting     tasty-hunit-0.10.1 (lib)
Building     tasty-hunit-0.10.1 (lib)
Building     tasty-quickcheck-0.10.2 (lib)
Installing   tasty-hunit-0.10.1 (lib)
Installing   tasty-quickcheck-0.10.2 (lib)
Completed    tasty-hunit-0.10.1 (lib)
Completed    tasty-quickcheck-0.10.2 (lib)
Configuring test suite 'psqueues-tests' for psqueues-0.2.7.3..
Preprocessing test suite 'psqueues-tests' for psqueues-0.2.7.3..
Building test suite 'psqueues-tests' for psqueues-0.2.7.3..
[ 1 of 15] Compiling Data.BitUtil     ( src/Data/BitUtil.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/BitUtil.o )
[ 2 of 15] Compiling Data.IntPSQ.Internal ( src/Data/IntPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/IntPSQ/Internal.o )

src/Data/IntPSQ/Internal.hs:66:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
   |
66 | import           Control.Applicative ((<$>), (<*>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/IntPSQ/Internal.hs:70:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
   |
70 | import           Data.Foldable       (Foldable)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/IntPSQ/Internal.hs:75:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Word’ is redundant
      except perhaps to import instances from ‘Data.Word’
    To import instances alone, use: import Data.Word()
   |
75 | import           Data.Word           (Word)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 3 of 15] Compiling Data.IntPSQ      ( src/Data/IntPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/IntPSQ.o )
[ 4 of 15] Compiling Data.OrdPSQ.Internal ( src/Data/OrdPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/OrdPSQ/Internal.o )
[ 5 of 15] Compiling Data.OrdPSQ      ( src/Data/OrdPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/OrdPSQ.o )
[ 6 of 15] Compiling Data.HashPSQ.Internal ( src/Data/HashPSQ/Internal.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/HashPSQ/Internal.o )

src/Data/HashPSQ/Internal.hs:59:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Data.Foldable’ is redundant
      except perhaps to import instances from ‘Data.Foldable’
    To import instances alone, use: import Data.Foldable()
   |
59 | import           Data.Foldable        (Foldable)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 7 of 15] Compiling Data.HashPSQ     ( src/Data/HashPSQ.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/HashPSQ.o )
[ 8 of 15] Compiling Data.PSQ.Class   ( tests/Data/PSQ/Class.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/PSQ/Class.o )

tests/Data/PSQ/Class.hs:17:19: warning: [GHC-39567] [-Wstar-is-type]
    Using ‘*’ (or its Unicode variant) to mean ‘Data.Kind.Type’
    relies on the StarIsType extension, which will become
    deprecated in the future.
    Suggested fix: Use ‘Type’ from ‘Data.Kind’ instead.
   |
17 | class PSQ (psq :: * -> * -> *) where
   |                   ^

tests/Data/PSQ/Class.hs:17:24: warning: [GHC-39567] [-Wstar-is-type]
    Using ‘*’ (or its Unicode variant) to mean ‘Data.Kind.Type’
    relies on the StarIsType extension, which will become
    deprecated in the future.
    Suggested fix: Use ‘Type’ from ‘Data.Kind’ instead.
   |
17 | class PSQ (psq :: * -> * -> *) where
   |                        ^

tests/Data/PSQ/Class.hs:17:29: warning: [GHC-39567] [-Wstar-is-type]
    Using ‘*’ (or its Unicode variant) to mean ‘Data.Kind.Type’
    relies on the StarIsType extension, which will become
    deprecated in the future.
    Suggested fix: Use ‘Type’ from ‘Data.Kind’ instead.
   |
17 | class PSQ (psq :: * -> * -> *) where
   |                             ^

tests/Data/PSQ/Class.hs:18:21: warning: [GHC-39567] [-Wstar-is-type]
    Using ‘*’ (or its Unicode variant) to mean ‘Data.Kind.Type’
    relies on the StarIsType extension, which will become
    deprecated in the future.
    Suggested fix: Use ‘Type’ from ‘Data.Kind’ instead.
   |
18 |     type Key psq :: *
   |                     ^
[ 9 of 15] Compiling Data.PSQ.Class.Util ( tests/Data/PSQ/Class/Util.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/PSQ/Class/Util.o )

tests/Data/PSQ/Class/Util.hs:14:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
   |
14 | import           Control.Applicative ((<$>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[10 of 15] Compiling Data.PSQ.Class.Gen ( tests/Data/PSQ/Class/Gen.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/PSQ/Class/Gen.o )

tests/Data/PSQ/Class/Gen.hs:12:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
   |
12 | import           Control.Applicative (pure, (<$>), (<*>))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[11 of 15] Compiling Data.PSQ.Class.Tests ( tests/Data/PSQ/Class/Tests.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/PSQ/Class/Tests.o )

tests/Data/PSQ/Class/Tests.hs:9:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
  |
9 | import           Control.Applicative   ((<$>))
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

tests/Data/PSQ/Class/Tests.hs:12:41: warning: [GHC-38856] [-Wunused-imports]
    The import of ‘Foldable’ from module ‘Data.Foldable’ is redundant
   |
12 | import           Data.Foldable         (Foldable, foldr)
   |                                         ^^^^^^^^
[12 of 15] Compiling Data.OrdPSQ.Tests ( tests/Data/OrdPSQ/Tests.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/OrdPSQ/Tests.o )
[13 of 15] Compiling Data.IntPSQ.Tests ( tests/Data/IntPSQ/Tests.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/IntPSQ/Tests.o )
[14 of 15] Compiling Data.HashPSQ.Tests ( tests/Data/HashPSQ/Tests.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Data/HashPSQ/Tests.o )
[15 of 15] Compiling Main             ( tests/Main.hs, /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests-tmp/Main.o )
[16 of 16] Linking /tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/build/psqueues-tests/psqueues-tests
Running 1 test suites...
Test suite psqueues-tests: RUNNING...
Test suite psqueues-tests: PASS
Test suite logged to:
/tmp/psqueues-0.2.7.3/dist-newstyle/build/x86_64-linux/ghc-9.8.1/psqueues-0.2.7.3/t/psqueues-tests/test/psqueues-0.2.7.3-psqueues-tests.log
1 of 1 test suites (1 of 1 test cases) passed.
```

--- a/psqueues.cabal
+++ b/psqueues.cabal
@@ -67,7 +67,7 @@ Library
 
     Build-depends:
           base     >= 4.2     && < 5
-        , deepseq  >= 1.2     && < 1.5
+        , deepseq  >= 1.2     && < 1.6
         , hashable >= 1.1.2.3 && < 1.5
 
     if impl(ghc>=6.10)
@@ -145,7 +145,7 @@ Test-suite psqueues-tests
     Build-depends:
           HUnit            >= 1.2 && < 1.7
         , QuickCheck       >= 2.7 && < 2.15
-        , tasty            >= 1.2 && < 1.5
+        , tasty            >= 1.2 && < 1.6
         , tasty-hunit      >= 0.9 && < 0.11
         , tasty-quickcheck >= 0.8 && < 0.11
 

